index.ts

```diff
@@ -0,0 +1,131 @@
+import { EventEmitter } from 'events';
+import { Request, Response } from 'express';
+
+interface Options {
+  isSerialized: boolean;
+  isCompressed?: boolean;
+  initialEvent?: string;
+}
+
+/**
+ * Server-Sent Event instance class
+ */
+export class SSE extends EventEmitter {
+  initial: string[];
+
+  /**
+   * Creates a new Server-Sent Event instance
+   * @param initial Initial value(s) to be served through SSE
+   * @param options SSE options
+   */
+  constructor(initial: any | any[] = [], public options: Options = { isSerialized: true }) {
+    super();
+
+    this.initial = Array.isArray(initial) ? initial : [initial];
+    this.init = this.init.bind(this);
+  }
+
+  /**
+   * The SSE route handler
+   */
+  init(request: Request, response: Response) {
+    let id = 0;
+    request.socket.setTimeout(0);
+    request.socket.setNoDelay(true);
+    request.socket.setKeepAlive(true);
+    response.statusCode = 200;
+    response.setHeader('Content-Type', 'text/event-stream');
+    response.setHeader('Cache-Control', 'no-cache');
+    response.setHeader('X-Accel-Buffering', 'no');
+    if (request.httpVersion !== '2.0') {
+      response.setHeader('Connection', 'keep-alive');
+    }
+    if (this.options.isCompressed) {
+      response.setHeader('Content-Encoding', 'deflate');
+    }
+
+    // Increase number of event listeners on init
+    this.setMaxListeners(this.getMaxListeners() + 2);
+
+    const dataListener = (data: any) => {
+      if (data.id) {
+        response.write(`id: ${data.id}\n`);
+      } else {
+        response.write(`id: ${id}\n`);
+        id += 1;
+      }
+      if (data.event) {
+        response.write(`event: ${data.event}\n`);
+      }
+      response.write(`data: ${JSON.stringify(data.data)}\n\n`);
+      response.flushHeaders();
```

Comment:
Not sure if this is correct or if this should be checking for if `response.flush` exists and then calling that.